doc/source/performance/xtradb_performance_improvements_for_io-bound_highly-concurrent_workloads.rst

```diff
@@ -46,6 +35,52 @@ The following |InnoDB| metrics are no longer accounted, as their semantics do no
 
 The need for |InnoDB| recovery thread writer threads is also removed, consequently all associated code is deleted.
 
+.. _parallel_doublewrite_buffer:
+
+Parallel doublewrite buffer
+===========================
+
+The existing doublewrite buffer is shared between all the buffer pool instances and all the flusher threads. PMP shows that it's becoming a point of contention with flusher threads waiting for the running batch to end so that they can post a page to flush. This limits the effect of extra cleaner threads. Moreover, single page flushes wait for the above and also contend on the doublewrite mutex.
```

Comment:
The legacy doublewrite buffer is shared between all the buffer pool instances and all the flusher threads. It collects all the page write requests into a single buffer, and, when the buffer fills, writes it out to disk twice, blocking any new write requests until the writes complete. This becomes a bottleneck with increased flusher parallelism, limiting the effect extra cleaner threads. In addition, single page flushes, if they are performed, are subject to above and also contend on the doublewrite mutex.